src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -128,18 +133,31 @@ public partial class AuthenticationResult
 
             if (homeAccountId != null)
             {
+                // This method call is to an async method from this constructor but the network call in the method will ideally never happen.
+                var tenantProfiles = Task.Run(async () => await _cacheManager.GetTenantProfilesAsync(homeAccountId).ConfigureAwait(false)).GetAwaiter().GetResult();
+
                 string username = null;
                 if (msalIdTokenCacheItem != null)
                 {
                     username = msalIdTokenCacheItem.IsAdfs ?
                         msalIdTokenCacheItem?.IdToken.Upn :
                         msalIdTokenCacheItem?.IdToken?.PreferredUsername;
+
+                    if (tenantProfiles != null)
+                    {
+                        tenantProfiles[msalIdTokenCacheItem.TenantId] = new TenantProfile(
+                            msalIdTokenCacheItem.IdToken.ObjectId,
+                            msalIdTokenCacheItem.TenantId,
+                            msalIdTokenCacheItem.Secret,
```

Comment:
>.Secret [](http://example.com/codeflow?start=48&length=7)

why use Secret and not `msalIdTokenCacheItem.IdToken`?

Reply:
Secret is the id token before parsing. We need this to get the claims. IdToken in the cache is the IdToken object [see](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/698f0fd1b0daba783c9028e099595f10a0d52da2/src/client/Microsoft.Identity.Client/Cache/Items/MsalIdTokenCacheItem.cs#L65)